src/graph.h

```diff
@@ -8,55 +8,65 @@
 #include <tuple>
 #include <list>
 #include <algorithm>
-#include <list>
-#include <climits>
-
-#include "UFDS.h"
+#include <memory>
 #include "vertex.h"
 #include "node.h"
 #include "UFDS.h"
 
 class Graph {
   private:
     unsigned int totalEdges = 0;
-    std::vector<Vertex *> vertexSet;    // vertex set
-    std::unordered_map<unsigned int, Vertex *> idToVertex;
-    std::pair<unsigned int, std::list<Vertex *>> tour;
+//?    std::unordered_map<unsigned int, std::shared_ptr<Vertex>> idToVertex;
+    struct tour_t {
+        unsigned int distance;
```

Comment:
This should be a long, as an Edge's length is a double

Reply:
It should be a double, shouldn't it?